src/plugins/AutoRefresh.ts

```diff
@@ -59,8 +61,10 @@ class AutoRefresh extends GenericPlugin {
       this.timeInView += 1;
 
       // Determine whether the ad should refresh
-      if (!this.isRefreshing && this.timeInView >= refreshRateInSeconds) {
+      if (!this.isRefreshing && this.timeInView >= refreshRateInSeconds
+        && (refreshLimit === null || this.refreshCount < refreshLimit)) {
```

Comment:
We should clear the interval when the limit is met

Reply:
Will do